tests/proptest.rs

```diff
@@ -18,19 +18,42 @@ use proptest::string::string_regex;
 mod sat_dependency_provider;
 use crate::sat_dependency_provider::SatResolve;
 
-/// The same as DP but takes versions from the opposite end:
-/// if DP returns versions from newest to oldest, this returns them from oldest to newest.
+/// The same as [OfflineDependencyProvider] but takes versions from the opposite end:
+/// if [OfflineDependencyProvider] returns versions from newest to oldest, this returns them from oldest to newest.
 #[derive(Clone)]
-struct ReverseDependencyProvider<DP>(DP);
-
-impl<P: Package, V: Version, DP: DependencyProvider<P, V>> DependencyProvider<P, V>
-    for ReverseDependencyProvider<DP>
-{
-    fn list_available_versions(&self, p: &P) -> Result<Vec<V>, Box<dyn Error>> {
-        self.0.list_available_versions(p).map(|mut v| {
-            v.reverse();
-            v
-        })
+struct OldestDependencyProvider<P: Package, V: Version>(OfflineDependencyProvider<P, V>);
```

Comment:
Isn't this `DepedencyProvider` rather new? :smile: 

Maybe `OldestVersionsDependencyProvider` would be better. Alternatively we leave it as `ReverseDependencyProvider` if we go with the suggestion to return newest versions first as the default trait implementation.